crates/subspace-farmer/src/farming.rs

```diff
@@ -58,35 +58,39 @@ impl Farming {
         // Oneshot channels, that will be used for interrupt/stop the process
         let (stop_sender, stop_receiver) = async_oneshot::oneshot();
 
+        let span = Span::current();
         // Get a handle for the background task, so that we can wait on it later if we want to
-        let farming_handle = tokio::spawn(async move {
-            match future::select(
-                Box::pin(subscribe_to_slot_info(
-                    single_plot_farm_id,
-                    &client,
-                    &plot,
-                    &commitments,
-                    single_disk_semaphore,
-                    &identity,
-                    reward_address,
-                )),
-                stop_receiver,
-            )
-            .await
-            {
-                Either::Left((farming_result, _)) => {
-                    if let Err(val) = farming_result {
-                        return Err(val);
+        let farming_handle = tokio::spawn(
+            async move {
+                match future::select(
+                    Box::pin(subscribe_to_slot_info(
+                        single_plot_farm_id,
+                        &client,
+                        &plot,
+                        &commitments,
+                        single_disk_semaphore,
+                        &identity,
+                        reward_address,
+                    )),
+                    stop_receiver,
+                )
+                .await
+                {
+                    Either::Left((farming_result, _)) => {
+                        if let Err(val) = farming_result {
+                            return Err(val);
+                        }
+                    }
+                    // either Sender is dropped, or a stop message is received.
+                    // in both cases, we stop the process and return Ok(())
+                    Either::Right((_, _)) => {
+                        info!("Farming stopped!");
                     }
                 }
-                // either Sender is dropped, or a stop message is received.
-                // in both cases, we stop the process and return Ok(())
-                Either::Right((_, _)) => {
-                    info!("Farming stopped!");
-                }
+                Ok(())
             }
-            Ok(())
-        });
+            .instrument(span),
```

Comment:
I believe you meant this?
https://docs.rs/tracing/0.1.35/tracing/trait.Instrument.html#method.in_current_span